### ISSUE TITLE
Need to prepare Python and C# (and maybe c++) api to prepare for 8.0 API changes

### ISSUE THREAD

**Issue by jackjansen** [cwi-dis/cwipc#220]
We need a `detach()` method on the point cloud objects (and other managed objects) to signal that we are going to pass it to another language.

That way, we can eventually let the Python or C# garbage collector handle our `free()` calls.

`pc.detach()` would return a new detached object (which will not free the underlying native object when freed) and it will invalidate the original object.
